.github/workflows/blackduck-scan.yaml

```diff
@@ -0,0 +1,109 @@
+# This central workflow is used to run a Black Duck SCA scan on a repository.
+# It is triggered by the Trigger Blackduck Scan workflow,
+# which is defined in the other repositories' .github/workflows
+name: Execute Blackduck SCA Scan
+
+on:
+  workflow_call:
+    inputs:
+      detector_search_depth:
+        description: 'Optional detector search depth for the Black Duck scan.'
+        required: false
+        type: string
+        default: '0'
+      pull_request:
+        description: 'Was the tiggering workflow done by a pull request?'
+        required: true
+        type: string
+    secrets:
+      BLACKDUCK_API_TOKEN:
+        required: true
+      BLACKDUCK_URL:
+        required: true
+
+permissions:
+  contents: read 
+
+jobs:
+  build:
+    runs-on: [ ubuntu-latest ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set Detector Search Depth (if provided)
+        if: ${{ inputs.detector_search_depth != '' }}
+        run: |
+          echo "DETECT_DETECTOR_SEARCH_DEPTH=${{ inputs.detector_search_depth }}" >> $GITHUB_ENV
```

Comment:
Why can we not use the value of `${{ inputs.detector_search_depth }}` directly? 

If the value of `${{ inputs.detector_search_depth }}` is empty, then nothing changes for the later set of `DETECT_DETECTOR_SEARCH_DEPTH`, does it?

Reply:
yes, corrected.